src/instructlab/model/train.py

```diff
@@ -520,20 +585,328 @@ def get_files(directory: str, pattern: str) -> list[str]:
+            phased_base_dir: str = str(kwargs.get("phased_base_dir"))
+            phased_phase1_data: str = str(kwargs.get("phased_phase1_data"))
+            phased_phase2_data: str = str(kwargs.get("phased_phase2_data"))
+            if all([phased_base_dir, phased_phase1_data, phased_phase2_data]):
+                if click.confirm(
+                    f"""\n\nSTARTING END-TO-END TRAINING. \nProceeding will OVERWRITE any metadata (checkpoints, intermediate results) that are stored at {phased_base_dir}.\n\nWould you like to proceed?""",
+                    abort=True,
+                ):
+                    _prepare_phased_base_dir(phased_base_dir)
+
+                    _run_phased_training(
+                        ctx=ctx,
+                        train_args=train_args,
+                        torch_args=torch_args,
+                        base_dir=phased_base_dir,
+                        phase1_data=phased_phase1_data,
+                        phase1_num_epochs=kwargs.get("phased_phase1_num_epochs"),
+                        phase1_samples_per_save=kwargs.get(
+                            "phased_phase1_samples_per_save"
+                        ),
+                        phase2_data=phased_phase1_data,
+                        phase2_num_epochs=kwargs.get("phased_phase2_num_epochs"),
+                        phase2_samples_per_save=kwargs.get(
+                            "phased_phase2_samples_per_save"
+                        ),
+                        mtbench_judge=str(kwargs.get("phased_mt_bench_judge")),
+                    )
+            else:
+                raise ctx.fail(
+                    "End-to-end training minimally requires paths to `base_dir`, `phase1_data`, and `phase2_data`."
+                )
+
+        else:
+            if not os.path.isfile(data_path):
+                ctx.fail(
+                    f"Data path must be to a valid .jsonl file. Value given: {data_path}"
+                )
+
+            try:
+                run_training(train_args=train_args, torch_args=torch_args)
+            # unsure what types of exceptions training library returns, this will catch all for now
+            except Exception as exc:
+                ctx.fail(f"Error while executing training library {exc}")
+
+
+def _prepare_phased_base_dir(phased_base_dir: str) -> None:
+    """Adds phase1 and phase2 directories and subdirectories in phased_base_dir.
+
+    Args:
+        phased_base_dir: directory wrapping phase1 and phase2 cached data.
+    """
+
+    logger.debug(f"Phased Training -- Preparing phased base dir: {phased_base_dir}")
+
+    phase1_dir_path = os.path.join(phased_base_dir, "phase1")
+    phase2_dir_path = os.path.join(phased_base_dir, "phase2")
+
+    _create_empty_directory(phase1_dir_path)
+    _create_empty_directory(phase2_dir_path)
+
+    _setup_phase_dirs(phase1_dir_path)
+    _setup_phase_dirs(phase2_dir_path)
+
+
+def _create_empty_directory(path: str) -> None:
```

Comment:
It would make sense to move this to utils IMO

Reply:
moved!